Alignment/OfflineValidation/macros/momentumElectronBiasValidation.C

```diff
@@ -158,6 +159,7 @@ void momentumBiasValidation(TString variable,
 
   // To save the table cut
   TFile* histo1;
+  delete histo1;
```

Comment:
What's the point if creating a pointer and immediately deleting it?

Reply:
Hi Marco! Yes, it's no point of this, but it helped to avoid a warning message which was caused at line 285. I wanted to create only commit to ask first, but I created a pull request by mistake. 
The idea was to solve the problem in the macros, which doesn't produce plots, because it exits the process after the "Initializing the TTree" step. This problem was solved by adding braces at lines 143 -146, but it caused a warning about "delete histo1" while compiling. So I moved it up and the warning disappeared and it didn't affect the final result.

Reply:
I don't see any point in making dynamic memory allocation in the first place, please provide the warning message.

Reply:
root [0] .L momentumElectronBiasValidation.C++
Info in <TUnixSystem::ACLiC>: creating shared library /afs/cern.ch/cms/CAF/CMSALCA/ALCA_TRACKERALIGN/data/commonValidation/alignmentObjects/dhohov/EoP2022/CMSSW_12_6_0_pre2/src/Alignment/OfflineValidation/macros/./momentumElectronBiasValidation_C.so
In file included from /afs/cern.ch/cms/CAF/CMSALCA/ALCA_TRACKERALIGN/data/commonValidation/alignmentObjects/dhohov/EoP2022/CMSSW_12_6_0_pre2/src/Alignment/OfflineValidation/macros/momentumElectronBiasValidation_C_ACLiC_dict.cxx:41:
/afs/cern.ch/cms/CAF/CMSALCA/ALCA_TRACKERALIGN/data/commonValidation/alignmentObjects/dhohov/EoP2022/CMSSW_12_6_0_pre2/src/Alignment/OfflineValidation/macros/./momentumElectronBiasValidation.C: In function 'void momentumBiasValidation(TString, TString, TString, TString, Double_t, Bool_t, Double_t, Double_t)':
/afs/cern.ch/cms/CAF/CMSALCA/ALCA_TRACKERALIGN/data/commonValidation/alignmentObjects/dhohov/EoP2022/CMSSW_12_6_0_pre2/src/Alignment/OfflineValidation/macros/./momentumElectronBiasValidation.C:285:10: warning: 'histo1' may be used uninitialized in this function [-Wmaybe-uninitialized]
  285 |   delete histo1;
      |          ^~~~~~
In file included from /afs/cern.ch/cms/CAF/CMSALCA/ALCA_TRACKERALIGN/data/commonValidation/alignmentObjects/dhohov/EoP2022/CMSSW_12_6_0_pre2/src/Alignment/OfflineValidation/macros/momentumElectronBiasValidation_C_ACLiC_dict.cxx:41:
/afs/cern.ch/cms/CAF/CMSALCA/ALCA_TRACKERALIGN/data/commonValidation/alignmentObjects/dhohov/EoP2022/CMSSW_12_6_0_pre2/src/Alignment/OfflineValidation/macros/./momentumElectronBiasValidation.C: In function 'void momentumBiasValidation(TString, TString, TString, TString, Double_t, Bool_t, Double_t, Double_t)':
/afs/cern.ch/cms/CAF/CMSALCA/ALCA_TRACKERALIGN/data/commonValidation/alignmentObjects/dhohov/EoP2022/CMSSW_12_6_0_pre2/src/Alignment/OfflineValidation/macros/./momentumElectronBiasValidation.C:285:10: warning: 'histo1' may be used uninitialized in this function [-Wmaybe-uninitialized]
  285 |   delete histo1;
      |          ^~~~~~
root [1]

Reply:
@dhohov in `CMSSW_13_0_X_2022-11-28-2300` (last master IB) I can't reproduce this warning.
```bash
cmsrel CMSSW_13_0_X_2022-11-28-2300/src/
cd CMSSW_13_0_X_2022-11-28-2300/src/
cmsenv
git cms-addpkg Alignment/OfflineValidation
scramv1 b -j 20
cd Alignment/OfflineValidation/macros/
root -b
   ------------------------------------------------------------------
  | Welcome to ROOT 6.24/07                        https://root.cern |
  | (c) 1995-2021, The ROOT Team; conception: R. Brun, F. Rademakers |
  | Built for linuxx8664gcc on Nov 26 2022, 05:31:00                 |
  | From tag , 2 September 2021                                      |
  | With g++ (GCC) 11.2.1 20210817                                   |
  | Try '.help', '.demo', '.license', '.credits', '.quit'/'.q'       |
   ------------------------------------------------------------------

root [0] .L momentumElectronBiasValidation.C++
Info in <TUnixSystem::ACLiC>: creating shared library /tmp/musich/CMSSW_13_0_X_2022-11-28-2300/src/Alignment/OfflineValidation/macros/./momentumElectronBiasValidation_C.so
root [1] .q

```

Reply:
Nor I can in `CMSSW_12_6_0_pre2` which is what https://github.com/cms-sw/cmssw/pull/40080#discussion_r1033410211 seems to imply

Reply:
If you use the original file, you will not have a warning, but you will not obtain plots too, which was a problem.
to run it: momentumBiasValidation("eta", "/afs/cern.ch/cms/CAF/CMSALCA/ALCA_TRACKERALIGN/data/commonValidation/alignmentObjects/..your_path_to_Eop_Trees/", "name_of_Treefile.root=TEST", "gif", true)

Reply:
https://github.com/cms-sw/cmssw/pull/40182 should solve. Please close this one.